### ISSUE TITLE
Regent: `__parallel_prefix` incorrect on Frontier

### ISSUE THREAD

**Issue by syamajala** [StanfordLegion/legion#1701]
We're seeing S3D diverge when moving a task to gpu. The task uses `__parallel_prefix` and it looks the output changes. I will try to find a minimal reproducer.

**Comment by syamajala** [comment]
@elliottslaughter i put a reproducer for this bug on frontier here: /lustre/orion/cmb103/world-shared/seshuy/prefix

**Comment by syamajala** [comment]
llvm 18 and rocm 6 do not fix this error. Think its a bug in Regent.

**Comment by elliottslaughter** [comment]
I have reproduced this on Frontier.

 * CPU version: passes
 * HIP version: fails
 * CUDA version: passes

Because the CUDA version passes, this is either a bug in HIP-specific code generation or something of a mismatch when applying generic algorithms to HIP.

**Comment by syamajala** [comment]
I am able to run tdb on frontier but the performance isnt super great. I think we need this.

**Comment by syamajala** [comment]
@elliottslaughter i think this is probably still there?

**Comment by syamajala** [comment]
I ran the standalone example and it seems to work on rocm 7.1.1 using llvm 20.

**Comment by elliottslaughter** [comment]
Ok, I guess if you have evidence of another failure in the full application, then we'd need to re-derive a new minimal reproducer. But with luck this will work now.

**Comment by syamajala** [comment]
Yeah it looks like it is still broken in S3D. Maybe I need to do more testing with the reproducer.
